src/codemagic/google/resource_managers/mixins/listing_manager_mixin.py

```diff
@@ -65,7 +65,9 @@ def list(
             request = self._get_resources_page_request(page_request_args)
             response = self._execute_request(request)
```

Comment:
The response might as well be `""` or even `None`
```suggestion
            response = self._execute_request(request) or {}
```

Reply:
For listing requests? If the response indeed is something other than what we expect now (a dictionary) then I find it to be better if the request handling fails and we'll become aware of the unexpected return type instead of blindly accepting all kinds of values.